modules/database/sqlite/activate.php

```diff
@@ -0,0 +1,164 @@
+					}
+				}
+				return $redirect_location;
+			}
+
+			// Get current basic setup data to install WordPress in the new DB.
+			$blog_title = get_bloginfo( 'name' );
+			$is_public  = (bool) get_option( 'blog_public' );
+			$language   = get_option( 'WPLANG' );
+			$user_email = get_option( 'admin_email' );
+			$admin_user = get_user_by( 'email', $user_email );
+			if ( ! $admin_user ) {
+				$admin_user = wp_get_current_user();
+			}
+
+			// If the current user is not the admin email user, look up the
+			// data for the current user. Additionally, attempt to keep them
+			// logged-in by retaining their current session. Depending on the
+			// site configuration, this is not 100% reliable as sites may store
+			// session tokens outside of user meta. However that does not lead
+			// to any problem, the user would simply be required to sign in
+			// again.
+			$current_user    = null;
+			$current_user_id = get_current_user_id();
+			if ( $current_user_id !== (int) $admin_user->ID ) {
+				$current_user = wp_get_current_user();
+			}
+			$user_sessions = get_user_meta( $current_user_id, 'session_tokens', true );
+
+			// Get current data to keep the Performance Lab plugin and relevant
+			// modules active in the new DB.
+			$active_plugins_option = array( plugin_basename( PERFLAB_MAIN_FILE ) );
+			$active_modules_option = get_option( PERFLAB_MODULES_SETTING, array() );
+
+			// Load and set up SQLite database.
+			require_once __DIR__ . '/wp-includes/sqlite/db.php';
+			wp_set_wpdb_vars();
+
+			// Load WordPress installation API functions.
+			require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+
+			// Install WordPress in the SQLite database with the same base
+			// configuration as the MySQL database.
+			// Since $admin_user->user_pass is already hashed, add a filter to
+			// ensure it is inserted into the database like that, instead of
+			// being re-hashed.
+			$unhash_user_pass = function( $data, $update, $user_id, $userdata ) use ( $admin_user, $current_user ) {
+				// Double check this is actually the already hashed password,
+				// to prevent any chance of accidentally putting another
+				// password into the database which would then be plain text.
+				if (
+					! empty( $userdata['user_pass'] )
+					&& (
+						$userdata['user_pass'] === $admin_user->user_pass
+						|| $current_user && $userdata['user_pass'] === $current_user->user_pass
+					)
+				) {
+					$data['user_pass'] = $userdata['user_pass'];
+				}
+				return $data;
+			};
+			add_filter( 'wp_pre_insert_user_data', $unhash_user_pass, 10, 4 );
+			wp_install( $blog_title, $admin_user->user_login, $user_email, $is_public, '', $admin_user->user_pass, $language );
+			if ( $current_user ) { // Also "copy" current admin user if it's not the admin email owner.
+				wp_create_user( $current_user->user_login, $current_user->user_pass, $current_user->user_email );
+			}
+			remove_filter( 'wp_pre_insert_user_data', $unhash_user_pass, 10 );
```

Comment:
10 is the default prio
```suggestion
			remove_filter( 'wp_pre_insert_user_data', $unhash_user_pass );
```